modules/gdnative/godot/godot_string.cpp

```diff
@@ -112,6 +112,359 @@ void GDAPI godot_string_destroy(godot_string *p_self) {
+    const String *self = (const String *) p_self;
+    
+    return self->length();
+}
+
+/* Helpers */
+
+godot_bool GDAPI godot_string_begins_with(const godot_string *p_self, const godot_string *p_string) {
+    const String *self = (const String *) p_self;
+    const String *string = (const String *) p_string;
+
+    return self->begins_with(*string);
+}
+
+godot_bool GDAPI godot_string_begins_with_char_array(const godot_string *p_self, const char *p_char_array) {
+    const String *self = (const String *) p_self;
+
+    return self->begins_with(p_char_array);
+}
+
+// /?\ How to convert a 'Vector<String>' to 'const Array&' ?
+/*
+godot_array GDAPI godot_string_bigrams(const godot_string *p_self;) {
+    godot_array bigrams;
+    const String *self = (const String *) p_self;
+    memnew_placement(&bigrams, Array(_self->bigrams()));
+
+    return bigrams;
+}
+*/
+
+godot_string GDAPI godot_string_chr(wchar_t p_character) {
+    godot_string result;
+    memnew_placement(&result, String(String::chr(p_character)));
+    
+    return result;
+}
+
+godot_bool GDAPI godot_string_ends_with(const godot_string *p_self, const godot_string *p_string) {
+    const String *self = (const String *) p_self;
+    const String *string = (const String *) p_string;
+
+    return self->ends_with(*string);
+}
+
+godot_int GDAPI godot_string_find(const godot_string *p_self, godot_string p_what) {
+    const String *self = (const String *) p_self;
+    String *what = (String *) &p_what;
+    
+    return self->find(*what);
+}
+
+godot_int GDAPI godot_string_find_from(const godot_string *p_self, godot_string p_what, godot_int p_from) {
+    const String *self = (const String *) p_self;
+    String *what = (String *) &p_what;
+    
+    return self->find(*what, p_from);
+}
+
+godot_int GDAPI godot_string_findn(const godot_string *p_self, godot_string p_what) {
+    const String *self = (const String *) p_self;
+    String *what = (String *) &p_what;
+    
+    return self->findn(*what);
+}
+godot_int GDAPI godot_string_findn_from(const godot_string *p_self, godot_string p_what, godot_int p_from) {
+    const String *self = (const String *) p_self;
+    String *what = (String *) &p_what;
+    
+    return self->findn(*what, p_from);
+}
+
+godot_int GDAPI find_last(const godot_string *p_self, godot_string p_what) {
+    const String *self = (const String *) p_self;
+    String *what = (String *) &p_what;
+    
+    return self->find_last(*what);
+}
+
+godot_string GDAPI godot_string_format(const godot_string *p_self, const godot_variant *p_values) {
+    const String *self = (const String *) p_self;
+    const Variant *values = (const Variant *) p_values;
+    godot_string result;
+    memnew_placement(&result, String(self->format(*values)));
+
+    return result;
+}
+
+godot_string GDAPI godot_string_format_with_custom_placeholder(const godot_string *p_self, const godot_variant *p_values, const char *p_placeholder) {
```

Comment:
The placeholder is a `String` in the C++ API. You can just use `godot_string` for that here too, all these functions will most likely be used via bindings which abstract the String type anyway.